proposal.rst

```diff
@@ -239,7 +337,99 @@ Project Timeline
 .. any university classes or exams, vacations, etc), make sure you include them
 .. in your project timeline.
 
-FILLME
+**Bonding stage: Prior–May 28**
+
+* Get to know the community and interact with mentors.
+* Keep diving into Xapian and Omega's code base.
+* Fix some issues to go through the code review process.
+* Read more papers and have a clearer understanding of different click models.
+* Have clear blueprint of the project and think about the rough implementation of click model abstract class.
+
+**Coding Week 1: May 28–June 3**
+
+* Discuss the details of abstract class of click model with mentors.
+* Start implementing the abstract class.
+
+**Coding Week 2: June 4–June 10**
+
+* Final Week in unversity.
+* Finish implementing the abstract class.
+* Adapt the exsiting SimplifiedDBN class to the abstract class.
+* Start working on the DBN class with EM algorithm.
+
+**Coding Week 3: June 11–June 17**
+
+* Final Week in university.
+* Finish the training method for the DBN class.
+* Discuss the DCM model template with mentors.
+* Start writing automated test for the click models.
+* **Deliverable:** DBN model with EM alogorithm
```

Comment:
Particularly for the first deliverable, you should build in some time to get it merged. Once you've got the first chunk merged, subsequent chunks are generally easier — and from experience, it can take a bit of back and forth on the first one. (While small contributions to Xapian ahead of time help, they don't iron out all the bumps in the process.)

I suggest you assume it'll take a couple of weeks, but can happen alongside other work.

Reply:
Okay. Besides this, I will try to create a PR for Xapian before the GSoC coding begins.

Reply:
> Besides this, I will try to create a PR for Xapian before the GSoC coding begins.

We really want to see students work through making a change, submitting a PR, and working through the review process before selection - if we see they can manage that, we can have more confidence about progress during the project (which typically involves quite a lot of patch review).  So to make a difference that needs to be in the next 2 weeks, though sooner is better.